akka-sse/src/main/scala/de/heikoseeberger/akkasse/ServerSentEventParser.scala

```diff
@@ -70,31 +122,34 @@ private final class ServerSentEventParser(maxEventSize: Int) extends GraphStage[
     import shape._
 
     setHandler(in, new InHandler {
-      private var lines = Vector.empty[String]
-      private var linesSize = 0L
-      override def onPush() = {
-        val (newLines, newSize) = grab(in) match {
-          case "" => // A server-sent event is terminated with a new line, i.e. an empty line
-            val (data, eventType, id, retry) = parseFields(lines)
-            if (data.isEmpty)
-              pull(in)
-            else
-              push(out, ServerSentEvent(
-                data.mkString(LF),
-                Option(eventType),
-                Option(id),
-                Option(retry).flatMap(silentlyToInt)
-              ))
-            (Vector.empty, 0L)
-          case line if linesSize + line.length > maxEventSize =>
-            failStage(new IllegalStateException(s"maxEventSize of $maxEventSize exceeded!"))
-            (Vector.empty, 0L)
-          case line =>
-            pull(in)
-            (lines :+ line, linesSize + line.length)
-        }
-        lines = newLines
-        linesSize = newSize
+
+      private val builder = new Builder()
+
+      override def onPush() = grab(in) match {
+        case "" => // A server-sent event is terminated with a new line, i.e. an empty line
+          if (builder.hasData) push(out, builder.build()) else pull(in)
+          builder.reset()
```

Comment:
No, wait, this is correct. We are talking about an event without data: that must be ignored.